%% BioMed_Central_Tex_Template_v1.06
%%                                      %
%  bmc_article.tex            ver: 1.06 %
%                                       %

%%IMPORTANT: do not delete the first line of this template
%%It must be present to enable the BMC Submission system to
%%recognise this template!!

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%          <8 June 2012>              %%
%%                                     %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.html and the instructions for   %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% additional documentclass options:
%  [doublespacing]
%  [linenumbers]   - put the line numbers on margins

%%% loading packages, author definitions

%\documentclass[twocolumn]{bmcart}% uncomment this for twocolumn layout and comment line below
\documentclass{bmcart}

%%% Load packages
%\usepackage{amsthm,amsmath}
%\RequirePackage{natbib}
%\RequirePackage[authoryear]{natbib}% uncomment this for author-year bibliography
%\RequirePackage{hyperref}
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %%
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %%
%%  submitted article.                         %%
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\def\VIKTOR#1{\medskip\par\noindent\textcolor{blue}{\bf VIKTOR: #1}\par\medskip}
\def\KIRIL#1{\medskip\par\noindent\textcolor{red}{\bf KIRIL: #1}\par\medskip}


\def\includegraphic{}
\def\includegraphics{}



%%% Put your definitions there:
\startlocaldefs
\endlocaldefs


%%% Begin ...
\begin{document}

%%% Start of article front matter
\begin{frontmatter}

\begin{fmbox}
\dochead{Research}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Converting academic papers in biodiversity science into computer knowledge}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Specify information, if available,       %%
%% in the form:                             %%
%%   <key>={<id1>,<id2>}                    %%
%%   <key>=                                 %%
%% Comment or delete the keys which are     %%
%% not used. Repeat \author command as much %%
%% as required.                             %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\author[
   addressref={aff1, aff2},                   % id's of addresses, e.g. {aff1,aff2}
   corref={aff1},                       % id of corresponding address, if any
  % noteref={n1},                        % id's of article notes, if any
   email={datascience@pensoft.net}   % email address
]{\inits{V.E.S.}\fnm{Viktor} \snm{Senderov}}

\author[
   addressref={aff3},                   % id's of addresses, e.g. {aff1,aff2}
                       % id of corresponding address, if any
  % noteref={n1},                        % id's of article notes, if any
   email={kivs@bultreebank.org}   % email address
]{\fnm{Kiril} \snm{Simov}}

\author[
   addressref={aff1, aff2},                   % id's of addresses, e.g. {aff1,aff2}
                        % id of corresponding address, if any
  % noteref={n1},                        % id's of article notes, if any
   email={penev@pensoft.net}   % email address
]{\fnm{Lyubomir} \snm{Penev}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%% Repeat \address commands as much as      %%
%% required.                                %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address[id=aff1]{%                           % unique id
  \orgname{Pensoft Publishers}, % university, etc
  \street{Prof. Georgi Zlatarski 12},                     %
  \postcode{1700}                                % post or zip code
  \city{Sofia},                              % city
  \cny{Bulgaria}                                    % country
}

\address[id=aff2]{%
  \orgname{Institute of Biodiversity and Ecosystems Research, Bulgarian Academy of Sciences},
  \street{},
  \postcode{}
  \city{Sofia},
  \cny{Bulgaria}
}

\address[id=aff3]{%
  \orgname{Institute of Information and Communication Technologies, Bulgarian Academy of Sciences},
  \street{},
  \postcode{}
  \city{Sofia},
  \cny{Bulgaria}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter short notes here                   %%
%%                                          %%
%% Short notes will be after addresses      %%
%% on first page.                           %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{artnotes}
%\note{Sample of title note}     % note to the article
%\note[id=n1]{Equal contributor} % note, connected to author
\end{artnotes}

\end{fmbox}% comment this for two column layout

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% Please refer to the Instructions for     %%
%% authors on http://www.biomedcentral.com  %%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstractbox}

\begin{abstract} % abstract
\parttitle{First part title} %if any
Text for this section.

\parttitle{Second part title} %if any
Text for this section.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The keywords begin here                  %%
%%                                          %%
%% Put each keyword in separate \kwd{}.     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{keyword}
\kwd{sample}
\kwd{article}
\kwd{author}
\end{keyword}

% MSC classifications codes, if any
%\begin{keyword}[class=AMS]
%\kwd[Primary ]{}
%\kwd{}
%\kwd[; secondary ]{}
%\end{keyword}

\end{abstractbox}
%
%\end{fmbox}% uncomment this for twcolumn layout

\end{frontmatter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% Please refer to the instructions for     %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%% start of article main body
% <put your article body there>

\section{Extended Abstract}

Semantic publishing is bringing about a revolution in the field of biological science. In the biomedical domain there are well-established efforts to extract information and discover knowledge from literature \cite{momtchev_expanding_2009, williams_open_2012, rebholz-schuhmann_facts_2005}. The biodiversity domain, and in particular biological systematics (from here on in this paper referred to as \emph{taxonomy}), is also going in the direction of semantization of its research outputs \cite{tzitzikas_integrating_2013,senderov_open_2016,penev_fast_2010}. In this contribution, we present our work to fill the missing gaps in ontology development in taxonomy, and to create a knowledge system based on taxonomic literature.

Taxonomy is a very old discipline dating back to possibly Aristotle, whose fundamental insight was to group living things in a hierarchy \cite{manktelow_history_2010}. The discipline took its modern from after Carl Linnaeus (1707 - 1778) \cite{manktelow_history_2010}. In his system, Linnaeaus proposed to group organisms into \emph{kingdoms, classes, orders, genera, and species}, give them scientific names, consisting of Latinized bionomials, list possible alternative names, and give a characteristic description of the groups \cite{bhl10277}. These groups are called taxa and give the name of the discipline.

Even though Linnaeus and his colleagues may have hoped to describe life on Earth during their lifetimes, we now know that there are millions of species still undiscovered and undescribed \cite{ratnasingham_dna-based_2013}. Moreover, with the advent of evolutionary biology the grouping principles have changed \cite{mallet_species_2001}. Therefore, the description of life on Earth is a perpetual process and cannot be completed with a single project that can then be converted into an ontology. Thus, our aim is to create an ontology not of biodiversity itself, but to create a more generic ontology of the taxonomic process. The ongoing use of this ontology will enable the formal description of the totality of biodiversity knowledge at any given point in time.

Taxonomy works by employing the scientific method. Researchers examine specimens and based on the phenotypic and genetic variation that they observe form a hypothesis \cite{deans_time_2012}. This hypothesis is called a taxon concept, a potential taxon, or---in the case of species-level taxa---a species hypothesis \cite{berendsohn_concept_1995}. A taxon concept describes the allowable phenotypic and genomic variation within a taxon by both listing which specimens belong to it, and defining its characters explicitly. It is a valid falsifiable scientific claim as it needs to fulfill certain verifiable evolutionary requirements. For example, a species level concept needs fit our current understanding of what a species is, and a concept at a higher taxonomic level needs to form a monophyletic group \cite{mallet_species_2001}. Taxon concepts are published in the treatment section \cite{catapano_taxpub:_2010} of a scholarly taxonomic paper. Our ontology models scholarly taxonomic papers, taxon concepts, Latinized names, and other entities important in the taxonomic process.

The publishing domain has been modeled through the Semantic Publishing and Referencing Ontologies (SPAR) \cite{peroni_semantic_2014}. Taxonomic articles in particular have been modeled through TaxPub \cite{catapano_taxpub:_2010}. Our bibliographic model has SPAR at its core with a few extensions that we've written to accommodate for TaxPub elements.

Occurrences of living things and sampling for specimens have previously been modeled through DarwinCore (DwC) and Darwin-Semantic Web (Darwin-SW) \cite{baskauf_darwin-sw:_2016, wieczorek_darwin_2012}. We incorporate the Darwin-SW model in our ontology as it is.

Latinized names have previously been modeled through the NOMEN ontology \cite{dmitriev_nomen_nodate} and partly through the Taxonomic Nomenclatural Status Terms (TNSS) \cite{morris_taxonomic_nodate}. Latinized names are governed by the International Code of Zoological Nomenclature (ICZN) \cite{iczn1999} and by the International Code of Nomenclature for algae, fungi, and plants (Melbourne Code) \cite{mcneill_international_2012}. While NOMEN and the TNSS take a top-down approach of modeling the Codes, we take a bottom-up approach of modeling the actual usage of taxonomic names in articles. Where possible we map the classes that we've defined to NOMEN.

Taxon concepts have been modeled in XML through the Biodiversity Information Standards (TDWG) Taxonomic Concept Transfer Schema (TCS) \cite{tdwg_taxon_concept}. A now defunct and unavailable taxon concept ontology had been previously developed \cite{devries_taxon_nodate}. As we wanted to incorporate all of the semantic features of taxon concepts as discussed in the concept taxonomy literature \cite{berendsohn_concept_1995, franz_perspectives:_2009, sterner_taxonomy_2017}, we've modeled taxon concepts in OWL de novo.

The OWL implementation of the OpenBiodiv ontology resides within a literate programming document \cite{rdf_guide}. A key innovation of the ontology is that it introduces the class Taxonomic Name Usage (TNU)\footnote{the term TNU has been discussed previously widely in the community and we do not claim credit for coining it} to connect bibliographic elements such as sections of the article, captions in a figure, etc. to taxonomic entities such as biological names. A taxonomic name usage in the text of a taxonomic article has the form "\emph{Binomial name} Authorship-Year, taxonomic status". An example would be "\emph{Heser stoevi} Deltschev, sp. n.". The taxonomic status at the end indicates the sense it which the scientific name has been invoked. The present example indicates a new species or "species novum." Although these taxonomic statuses are governed by the codes, in their actual usage, there is a considerable variation in spelling, and even in the sense that they are employment. We've investigated the actual use of these statuses in around 4,000 articles published across four journals (zookeys,...) and came up with taxonomic status vocabulary /cite{ ...} Furthermore, the validity of Latinized names can be inferred automatically with the help of the taxonomic name usages /cite{rules}. The related-ness of names can be also inferred on the basis of the co-occurrence in a specific part of the manuscript. Another approach to infer related names, to which our approach will be compared, is the use of distributional semantics.

We've introduced a \emph{taxon concept} class as an equivalent class to DarwinCore's taxon, as we believe their definitions to be intensionally the same. Our ontology allows for multiplicity of opinion, i.e. multiple coexisting taxonomies according to different sources /cite{something by franz}. Furthermore, we define a way to express not only simple parent-child relationship between taxon concepts, but also relationships of partial overlap, disjointness, equality, proper inclusion, and inverse proper inclusion, both intensively and intensionally as defined by the Region Connection Calculus-5 /cite{ }

Our ontology servers as the basis for the Open Biodiversity Knowledge Management System (OBKMS, or OpenBiodiv) \cite{Senderov2016}. Currently, the system is under development with millions of triples and thousands of articles already having been processed. The usefulness of the ontology will be determined by how well it serves the needs of OpenBiodiv. One of its strong-points is that by filling the ontological gaps it allows the creation of a biodiversity knowledge graph /cite{page}, and linking all taxonomic information together: occurrences, specimens, names, taxon concepts, articles, treatments, images, etc. This will allow the users of the system to answer competency questions /cite{give a link to comptency questions article}...

Furthermore, by using taxon concepts and taxon concept labels, the system a and avoids some of the problems of having a consensus (backbone taxonomy)


%%%%%%%%%%%%%%%%
%% Background %%
%%
\section{Introduction}
%n this section I will state the stage for the problem.%

Semantic publishing is bringing about a revolution in the field of biological sciences. Bioinformaticians, working in the fields of biodiversity science, as well as in the related fields of biomedicine, genomics, systems biology, etc., have been looking for ways to make available for machine computation the data locked up in research papers by pushing it into databases. Rebholz-Schuhmann et al. (TODO perhaps others, openphacts?) \cite{ } discussed the application of text-mining algorithms to extract information from biomedical literature.

Simultaneously, biological systematistists have been frustrated by the lack of an integrated information system serving the needs of the community \cite{}. As the field of biological systematics is very old, dating back Linnaeus (1756) \cite{}, there is a huge amount of legacy literature that ought to be processed via text-mining, if this integrated system is to be completed (TODO cite DONAT). At the same new species are being discovered and described daily, as we have far from described all of the existing biodiversity on Earth \cite{}. 

Text descriptions of new species are very structured \cite{}. This has lead to some interesting developments in the field of semantic publishing  \cite{}. 

A revolution is occurring enabling authors who publish in semantic journals---such as in Biodiversity Data Journal---to have their paper published both as a human-readable manuscript and machine-readable data. At present, upon publication the author has both a manuscript in the traditional PDF/HTML variants, an XML document, and, parts of the paper containing information about species occurrence records, images, taxonomic information, etc., are processed and deposited in international repositories such as GBIF, Zenodo, or Plazi.

The purpose of the present work to push this effort forward by creating an ontology to support information extraction from biodiversity literature. It will enable the linking of heterogeneous information from various sources in a unified system called the OBKMS. It will also convert the information into knowledge by introducing semantics enabling for the automatic evaluation of competency questions.

\section{Background}

\subsection{Domain Description}

%\KIRIL{Comment 1}

%\VIKTOR{Comment 2}

The purpose of this paper is to introduce an ontology for information extraction and knowledge discovery from taxonomic and biodiversity publications. Thus, the universe of discourse are taxonomic and biodiversity publications and the information contained therein. In this section we describe our understanding of the universe of discourse. We refer to this understanding as conceptualization.

There are two fundamental families in which the modeled entities fall. The first family is that of the bibliographic entities such as the article and its components. Examples would be the title, the abstract, the list of authors, various tables, figures and so on. The second family is that of the entities that make up the information content of the bibliographic elements. Examples include the persons that are in the list of authors, the biological organisms being discussed, and the occurrences of these organisms. Furthermore, it is not possible to constraint the list of non-bibliographic entities that a taxonomic or biodiversity article could in principle discuss. Various neighboring disciplines such as geography, medicine, or agriculture might be the subject matter. In the remainder of the section we will present our conceptualization of bibliographic and non-bibliographic entities specific to a taxonomic or biodiversity publication.

A taxonomic article adheres to the structure of a research article with a few exceptions. To understand the structural elements of a taxonomic article, we will now introduce the basic process by which taxonomy works.

The goal of taxonomy is to describe in a systematic way the biodiversity on Earth. Although descriptive at first glance, taxonomy works by employing the scientific method. The basic hypothesis that a taxonomist forms is that a specimen under examination belongs to a named group of organisms (taxon, pl. taxa) \cite{deans_time_2012}. This hypothesis is called a taxon concept, a potential taxon, or in the case of species-level taxa, a species hypothesis \cite{Berendsohn1995}. A taxon needs to fulfill certain evolutionary requirements such as to conform to a species concept or form a monophyletic group \cite{mallet_species_2001} and thus every taxon hypothesis is a falsifiable claim.

Taxa form a hierarchy---called taxonomy---that has given its name to the discipline. At the bottom of the hierarchy usually, but not always, is the species. Each organism is first a member of its species, then its genus, then its family, and so on. The top level taxa are kingdoms and domains. These hierarchy levels are referred to as ranks and are not completely fixed. The usage of lower ranks is governed by international codes \cite{W.D.L.Ride2012, McNeill2011}. Which specific ranks a given taxonomic paper employs is therefore dependent on the field (e.g. botany vs zoology), on the particular author, as well as on the level of taxonomic resolution required.

Taxon concepts are described in a specialized section called taxonomic treatment \cite{catapano_taxpub:_2010}. The treatment begins by introducing a valid Latinized name for the taxon. The Latinized name is accompanied by a taxonomic status, also usually in Latin, specifying whether a previously unknown taxon is being described, a known taxon is being redescribed, or a name of a taxon is being changed. Name changes may reflect a new position in the hierarchy (e.g. a species goes to a different genus) or for purely nomenclatural reasons (e.g. a spelling error is corrected, homonymy has been discovered). In addition to the name and the taxonomic status of the discussed taxon concept, related taxon concepts are listed in the nomenclature section of the treatment. The rules governing the use Latinized names are to be found in the Codes of Zoological or Botanical Nomenclature \cite{W.D.L.Ride2012,McNeill2011}.

After the nomenclatural subsection, a materials subsection is found within the treatment, where the examined specimens on which the taxon concept is based upon are listed in a highly structured manner. Each individual observation is called an occurrence record.

A further diagnosis section is found in the treatment as well that describes the traits, i.e. the range of allowable phenotypic variation for a taxon \cite{deans_time_2012}. Also we may have sections containing notes, biogeographic information, discussion and so on. Besides the treatment, another important section of a taxonomic article is the key, where one may find a step-by-step procedure for the classification of specimens. The TaxPub schema \cite{Catapano, penev_implementation_2012} provides a full XML-based formal description of the structure taxonomic article.

Recent research has shown that Latinized do not always unambiguously identify the concept that the author refers to \cite{remsen_use_2016,franz_names_2016,patterson_names_2010}. Rather, names identify lineages of evolving meaning over time, or even potentially coexisting concepts \cite{sterner_taxonomy_2017}. In order to eliminate this subjective step as it cannot be carried out by machines, we have modeled a particular type of biological name, called taxon concept label that includes a reference to the taxon concept in addition to the biological name \cite{franz_logic_2016}.

The use of taxon concept labels instead of traditional Latinized names has spawned a new branch of taxonomy called concept taxonomy. Concept taxonomists use five properties from the region connection calculus (RCC-5), namely equal, partially overlapping, properly included, inversely properly included and disjoint to denote relations between taxon concepts \cite{franz_two_2016}.

\subsection{Previous Work}

This paper is part of the project to build the Open Biodiversity Knowledge Management System \cite{Senderov2016}.

Gruber \cite{Gruber1993} lays the foundation of our understanding of \emph{ontology} as a specification of a conceptualization. Obitko, and Guarino et al. \cite{Obitko2007,Guarino2009}, further explore the foundations of ontologies.

The publishing domain has been modeled with the Semantic Publishing and Referencing Ontologies (SPAR) by Peroni \cite{Peroni2014}.  In addition to the resources from SPAR, we also utilize resources from the Treatment Ontologies of Catapano et. al \cite{Catapano} to model the bibliographic elements of a biodiversity publication.

There are several projects that are aimed at modeling the biodiversity domain. Baskauf and Webb \cite{Baskauf2014} adapt the previously existing DarwinCore data model by Wieczorek et al. \cite{Wieczorek2012} to RDF. These models deal primarily with occurrence data.

Biological names have previously been modeled in NOMEN by Dmitriev and Yoder \cite{Dmitriev} and partly by the Taxonomic Nomenclatural Status Terms of Morris and Morris \cite{Morris}. Biological names are governed by the Codes of Zoological and Botanical Nomenclature \cite{W.D.L.Ride2012,McNeill2011}. Biological names pose numerous challenges to systematics researchers \cite{Franz2014} and in this data model enhanced by taxon concepts and taxon concept labels.

Taxon concepts have been introduced by Berendsohn \cite{Berendsohn1995} and been discussed extensively by Franz and Sterner \cite{Franz2008,Sterner2017}. Taxon concepts have been modeled as XML in the Taxonomic Concept Transfer Schema \cite{Group2006}. A now defunct ontology exists, modeling taxon concepts as RDF \cite{DeVries2013}.


To combine the various concepts together we rely on several top-level ontologies such as PROTON \cite{Damova,Base2003} and SKOS \cite{Miles2009}. 

\cite{Obitko2007}

\cite{Guarino2009}



\section*{Ontology}

This is the formal introduction of the ontology, i.e. how the domain description is formalized with the help the help of various modeling techniques. More figures.

\subsection*{Integration of Previously Available Ontologies}

Here I discuss which previously existing ontologies or parts thereof (and why) are incorporated into the  OpenBiodiv Ontology.

\subsection*{Ontology Proper}

Here, I discuss the unique content of the OpenBiodiv Ontology. This section is further subdivided into subsubsections based on the topic of the classes and roughly follows the RDG Guide.

\subsubsection*{Subject Classification Vocabularies}
\subsubsection*{Taxon Name Usages}
\subsubsection*{Taxon Names}
\subsubsection*{Taxon Concepts}
\subsubsection*{Taxon Status Vocabulary}

\section*{Discussion}

In this section I discuss how the ontology can be used for information extraction from scholarly biodiversity papers, and what reasoning (with the help of additional SPARQL) rules is available from the ontology in order to answer scientific questions.

\section*{Conclusions}

Future work, etc.

\section*{TO DELETE}

Text for this section \ldots
\subsection*{Sub-heading for section}
Text for this sub-heading \ldots
\subsubsection*{Sub-sub heading for section}
Text for this sub-sub-heading \ldots
\paragraph*{Sub-sub-sub heading for section}
Text for this sub-sub-sub-heading \ldots
In this section we examine the growth rate of the mean of $Z_0$, $Z_1$ and $Z_2$. In
addition, we examine a common modeling assumption and note the
importance of considering the tails of the extinction time $T_x$ in
studies of escape dynamics.
We will first consider the expected resistant population at $vT_x$ for
some $v>0$, (and temporarily assume $\alpha=0$)
%
\[
 E \bigl[Z_1(vT_x) \bigr]= E
\biggl[\mu T_x\int_0^{v\wedge
1}Z_0(uT_x)
\exp \bigl(\lambda_1T_x(v-u) \bigr)\,du \biggr].
\]
%
If we assume that sensitive cells follow a deterministic decay
$Z_0(t)=xe^{\lambda_0 t}$ and approximate their extinction time as
$T_x\approx-\frac{1}{\lambda_0}\log x$, then we can heuristically
estimate the expected value as
%
\begin{eqnarray}\label{eqexpmuts}
E\bigl[Z_1(vT_x)\bigr] &=& \frac{\mu}{r}\log x
\int_0^{v\wedge1}x^{1-u}x^{({\lambda_1}/{r})(v-u)}\,du
\nonumber\\
&=& \frac{\mu}{r}x^{1-{\lambda_1}/{\lambda_0}v}\log x\int_0^{v\wedge
1}x^{-u(1+{\lambda_1}/{r})}\,du
\nonumber\\
&=& \frac{\mu}{\lambda_1-\lambda_0}x^{1+{\lambda_1}/{r}v} \biggl(1-\exp \biggl[-(v\wedge1) \biggl(1+
\frac{\lambda_1}{r}\biggr)\log x \biggr] \biggr).
\end{eqnarray}
%
Thus we observe that this expected value is finite for all $v>0$ (also see \cite{koon,khar,zvai,xjon,marg}).
%\nocite{oreg,schn,pond,smith,marg,hunn,advi,koha,mouse}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Backmatter begins here                   %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{backmatter}

\section*{Competing interests}
  The authors declare that they have no competing interests.

\section*{Author's contributions}
    Text for this section \ldots

\section*{Acknowledgements}
  Text for this section \ldots
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%
%%  Bmc_mathpys.bst  will be used to                       %%
%%  create a .BBL file for submission.                     %%
%%  After submission of the .TEX file,                     %%
%%  you will be prompted to submit your .BBL file.         %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %%
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %%
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% if your bibliography is in bibtex format, use those commands:
\bibliographystyle{bmc-mathphys} % Style BST file (bmc-mathphys, vancouver, spbasic).
\bibliography{bmc_article}      % Bibliography file (usually '*.bib' )
% for author-year bibliography (bmc-mathphys or spbasic)
% a) write to bib file (bmc-mathphys only)
% @settings{label, options="nameyear"}
% b) uncomment next line
%\nocite{label}

% or include bibliography directly:
% \begin{thebibliography}
% \bibitem{b1}
% \end{thebibliography}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}
  \begin{figure}[h!]
  \caption{\csentence{Sample figure title.}
      A short description of the figure content
      should go here.}
      \end{figure}

\begin{figure}[h!]
  \caption{\csentence{Sample figure title.}
      Figure legend text.}
      \end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
\begin{table}[h!]
\caption{Sample table title. This is where the description of the table should go.}
      \begin{tabular}{cccc}
        \hline
           & B1  &B2   & B3\\ \hline
        A1 & 0.1 & 0.2 & 0.3\\
        A2 & ... & ..  & .\\
        A3 & ..  & .   & .\\ \hline
      \end{tabular}
\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Additional Files}
  \subsection*{Additional file 1 --- Sample additional file title}
    Additional file descriptions text (including details of how to
    view the file, if it is in a non-standard format or the file extension).  This might
    refer to a multi-page table or a figure.

  \subsection*{Additional file 2 --- Sample additional file title}
    Additional file descriptions text.


\end{backmatter}
\end{document}
